backend/controller/internal/dal/dal.go

```diff
@@ -354,6 +355,22 @@ func (d *DAL) CreateDeployment(ctx context.Context, language string, schema *sch
 	if err != nil {
 		return "", errors.Wrap(err, "could not start transaction")
 	}
+
+	// This defer must be before the commit/rollback defer.
+	defer func() {
+		if err == nil {
+			err = d.InsertDeploymentEvent(ctx, &DeploymentEvent{
+				DeploymentName: key,
+				Time:           time.Now(),
+				Type:           DeploymentCreated,
+				Language:       language,
+				ModuleName:     schema.Name,
+			})
+			if err != nil {
+				fmt.Printf("failed to insert deployment event: %s\n", err)
+			}
+		}
+	}()
```

Comment:
@alecthomas is this approach ok? I needed the previous transaction to be complete before I could add an event since it's going to look up the `deployment_id` from the `key`

Reply:
I thought the DAL was going to insert the deployment events directly, in a single transaction?

Reply:
If the DAL doesn't do it, there's some chance that the event won't get inserted.

Reply:
Isn't this the dal?

Reply:
Sorry I didn't see the filename, and it was a bit confusing because this isn't in the transaction?

Reply:
And I don't quite understand why this needs to be in a defer? Why not just put it directly in the function body?